librz/core/casm.c

```diff
@@ -64,6 +64,108 @@ RZ_API char *rz_core_asm_search(RzCore *core, const char *input) {
 	return ret;
 }
 
+static const char *has_esil(RzCore *core, const char *name) {
+	RzListIter *iter;
+	RzAnalysisPlugin *h;
+	rz_return_val_if_fail(core && core->analysis && name, NULL);
+	rz_list_foreach (core->analysis->plugins, iter, h) {
+		if (h->name && !strcmp(name, h->name)) {
+			return h->esil ? "Ae" : "A_";
+		}
+	}
+	return "__";
+}
+
+RZ_API RzCmdStatus rz_core_asm_plugins_print(RzCore *core, const char *arch, RzCmdStateOutput *state) {
+	int i;
+	const char *feat2, *feat;
+	RzAsm *a = core->rasm;
+	char bits[32];
+	RzAsmPlugin *h;
+	RzListIter *iter;
+	PJ *pj = 0;
+	if (arch) {
+		rz_list_foreach (a->plugins, iter, h) {
+			if (h->cpus && !strcmp(arch, h->name)) {
+				char *c = strdup(h->cpus);
+				int n = rz_str_split(c, ',');
+				for (i = 0; i < n; i++) {
+					rz_cons_println(rz_str_word_get0(c, i));
+				}
+				free(c);
+				break;
+			}
+		}
+	} else {
+		if (state->mode == RZ_OUTPUT_MODE_JSON) {
+			pj = state->d.pj;
+			rz_cmd_state_output_array_start(state);
+		}
+		rz_list_foreach (a->plugins, iter, h) {
+			bits[0] = 0;
+			/* The underscore makes it easier to distinguish the
+			 * columns */
+			if (h->bits & 8) {
+				strcat(bits, "_8");
+			}
+			if (h->bits & 16) {
+				strcat(bits, "_16");
+			}
+			if (h->bits & 32) {
+				strcat(bits, "_32");
+			}
+			if (h->bits & 64) {
+				strcat(bits, "_64");
+			}
+			if (!*bits) {
+				strcat(bits, "_0");
+			}
+			feat = "__";
+			if (h->assemble && h->disassemble) {
+				feat = "ad";
+			}
+			if (h->assemble && !h->disassemble) {
+				feat = "a_";
+			}
+			if (!h->assemble && h->disassemble) {
+				feat = "_d";
+			}
+			feat2 = has_esil(core, h->name);
+			const char *license = h->license
+				? h->license
+				: "unknown";
+			switch (state->mode) {
+			case RZ_OUTPUT_MODE_QUIET: {
+				rz_cons_println(h->name);
+				break;
+			}
+			case RZ_OUTPUT_MODE_JSON: {
+				pj_ko(pj, h->name);
+				pj_ks(pj, "bits", bits);
+				pj_ks(pj, "license", license);
+				pj_ks(pj, "description", h->desc);
+				pj_ks(pj, "features", feat);
+				pj_end(pj);
+				break;
+			}
+			case RZ_OUTPUT_MODE_STANDARD: {
+				rz_cons_printf("%s%s  %-9s  %-11s %-7s %s\n",
+					feat, feat2, bits, h->name, license, h->desc);
+				break;
+			}
+			default: {
+				rz_warn_if_reached();
+				return RZ_CMD_STATUS_NONEXISTINGCMD;
+			}
+			}
+		}
+		if (state->mode == RZ_OUTPUT_MODE_JSON) {
+			rz_cmd_state_output_array_end(state);
```

Comment:
same here. Just use the `rz_cmd_state_output_array_end` function without the `if`.